### PR TITLE
Shim package:test to avoid matcher issues

### PR DESCRIPTION
**Pull request by Hixie**
cc @tvolkert @jonahwilliams 

~~This isn't quite done yet (I haven't tested by `test.dart` change and I need to remove some more dependencies on `package:test`) but I figured I'd give you a head start on reviewing it.~~

### CODE REVIEW

File: dev/bots/test.dart [hunk 1]
```diff
@@ -528,6 +529,33 @@ Future<Null> _runFlutterAnalyze(String workingDirectory, {
   );
 }
 
+Future<Null> _verifyNoTestPackageImports(String workingDirectory) async {
```
Comment (tvolkert) [inline dev/bots/test.dart]: Why not put this in `flutter_tools/test/forbidden_imports_test.dart`?

Comment (Hixie) [reply dev/bots/test.dart]: That only checks `flutter_tools`. This checks the whole repo.

File: dev/bots/test.dart [hunk 2]
```diff
@@ -528,6 +529,33 @@ Future<Null> _runFlutterAnalyze(String workingDirectory, {
...
+
+  // Fail if any errors
+  if (errors.isNotEmpty) {
+    print('$red━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━$reset');
+    final String s = errors.length == 1 ? '' : 's';
+    print('${bold}The following file$s depend on \'package:test\' directly:$reset\n');
+    print('\n');
+    print(errors.map((String line) => '  $line').join('\n'));
+    print('\n');
+    print('Rather than depending on \'package:test\' directly, use the shims provided.\n');
```
Comment (jonahwilliams) [inline dev/bots/test.dart]: Could you list the locations of the shims in the error message?

Comment (Hixie) [reply dev/bots/test.dart]: done

**Comment by Hixie** [comment]
Ok assuming the bots go green this should be good to go.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Review by tvolkert — APPROVED** [review]
LGTM

**Pull request merged**